invenio_jobs/jobs.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Jobs is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Jobs module."""
+
+from functools import partial
+
+
+class RegisteredTask:
```

Comment:
Simply `Task` to make it shorter? In `invenio-vocabularies`, the class name + inheritance becomes very long.

I am a bit confused by the naming: it has inside a `task` param.
Wasn'it that we have a `Job`, and a `Job Run` (its instance)? And the task is the associated celery task to execute?

And I would improve the doc string: I would explain what is it, and also it is not only for the admin panel, but in general a declarative way of defining how it is called, even programmatically.

Reply:
It is a good point,  thanks, in general I was struggling a bit with naming here. I want to highlight the fact that it is something that is not actually running, but something to be registered as an option. Also Task is repeated in celery context, so I want to come up with a name that is explicit enough to cover everything. Maybe you have some more ideas we could discuss?